src/cast_core/test/SampleHttpClient_test.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bsoncxx/json.hpp>
+#include <bsoncxx/builder/stream/document.hpp>
+
+#include <boost/exception/diagnostic_information.hpp>
+
+#include <yaml-cpp/yaml.h>
+
+#include <testlib/MongoTestFixture.hpp>
+#include <testlib/ActorHelper.hpp>
+#include <testlib/helpers.hpp>
+
+#include <gennylib/context.hpp>
+
+namespace genny {
+namespace {
+using namespace genny::testing;
+namespace bson_stream = bsoncxx::builder::stream;
+
+//
+// ⚠️ There is a "known" failure that you should find and fix as a bit of
+// an exercise in reading and testing your Actor. ⚠️
+//
+
+TEST_CASE_METHOD(MongoTestFixture, "SampleHttpClient successfully connects to a MongoDB instance.",
+          "[standalone][single_node_replset][three_node_replset][sharded][SampleHttpClient]") {
+
+    dropAllDatabases();
+    auto db = client.database("mydb");
+
+    NodeSource nodes = NodeSource(R"(
```

Comment:
Let's remove this file for now since the new actor doesn't have an associated workload. We can sync up with stakeholders first to see what kind of unit testing makes sense.

It may be tricky to write a test based on this template since this runs on Windows by default.